aws/validators_test.go

```diff
@@ -972,6 +972,83 @@ func TestValidateSQSFifoQueueName(t *testing.T) {
 	}
 }
 
+func TestValidateSNSNonFifoTopicName(t *testing.T) {
+	validNames := []string{
+		"valid-name",
+		"valid02-name",
+		"Valid-Name1",
+		"_",
+		"-",
+		strings.Repeat("W", 250),
+	}
+	for _, v := range validNames {
+		if errors := validateSNSNonFifoTopicName(v); len(errors) > 0 {
+			t.Fatalf("%q should be a valid SNS topic Name", v)
+		}
+	}
+
+	invalidNames := []string{
+		"Here is a name with: colon",
+		"another * invalid name",
+		"also $ invalid",
+		"This . is also %% invalid@!)+(",
+		"*",
+		"",
+		" ",
+		".",
+		strings.Repeat("W", 257), // length > 256
+	}
+	for _, v := range invalidNames {
+		if errors := validateSNSNonFifoTopicName(v); len(errors) == 0 {
+			t.Fatalf("%q should be an invalid SNS topic Name", v)
+		}
+	}
+}
+
+func TestValidateSNSFifoTopicName(t *testing.T) {
+	validNames := []string{
+		"valid-name.fifo",
+		"valid02-name.fifo",
+		"Valid-Name1.fifo",
+		"_.fifo",
+		"a.fifo",
+		"A.fifo",
+		"9.fifo",
+		"-.fifo",
+		fmt.Sprintf("%s.fifo", strings.Repeat("W", 250)),
```

Comment:
Confirmed, according to the API docs, all FIFO topics must end with the suffix of `.fifo` in the name.

https://docs.aws.amazon.com/sns/latest/api/API_CreateTopic.html